app/cms/utils/hooks-network-status.tsx

```diff
@@ -0,0 +1,29 @@
+import { useState } from "react"
+import useSWR from "swr"
+import { StatuspageApiResponse } from "~/ui/design-system/src/lib/interfaces"
+import { STATUSPAGE_API_KEY, STATUSPAGE_API_URL } from "./constants"
+
+export default function useNetworkStatuses() {
+  const [networkStatus, setNetworkStatus] = useState<StatuspageApiResponse[]>(
+    []
+  )
+
+  useSWR(
+    `${STATUSPAGE_API_URL}`,
+    (url) =>
+      fetch(url, {
+        headers: {
+          Authorization: `OAuth ${STATUSPAGE_API_KEY}`,
+        },
+      }).then((r) => r.json()),
+    {
+      refreshInterval: 100000,
+      onSuccess: (result) => {
+        console.log("setting from result")
+        setNetworkStatus(result)
+      },
+    }
+  )
+  console.log("network status", networkStatus)
+  return networkStatus as StatuspageApiResponse[]
+}
```

Comment:
@codingone21 i would recommend avoiding the `swr` dependency in a remix app: instead recommend the loaders and `useFetcher` if you need to do things like polling. this is desirable for a few reason:

- you won't have CORS issues since the data will come from remix (i.e. the same origin), and remix loaders will make the request from it's backend
- data will be server rendered, not fetched after page load, which is in the spirit of remix and is faster


data fetching things like swr, react-query, graphql etc aren't needed in remix apps

Reply:
i don't know if this is the best way to implement polling in remix, but something like this is what i'm trying to suggest:

https://benborgers.com/posts/remix-poll